examples/wandb_sweeps/train_wandb_sweep_agent.py

```diff
@@ -0,0 +1,140 @@
+import functools
+import os
+import traceback
+
+import pytorch_lightning as pl
+import wandb
+
+from yoyodyne import (
+    collators,
+    dataconfig,
+    defaults,
+    models,
+    predict,
+    schedulers,
+    train,
+    util,
+)
+
+
+class Error(Exception):
+    pass
+
+
+def run_train(args):
+    # First get trainer to initialize the wandb run
+    trainer = train._get_trainer_from_argparse_args(args)
+    pl.seed_everything(args.seed)
+    train_set, dev_set = train._get_datasets_from_argparse_args(args)
+    index = train.get_index(args.model_dir, args.experiment)
+    train_set.index.write(index)
+    util.log_info(f"Index: {index}")
+
+    # Model args come from the W&B sweep config.
+    kwargs = dict(wandb.config)
+    # Anything not specified in the config is taken from the CLI args.
+    kwargs.update({k: v for k, v in vars(args).items() if k not in kwargs})
+    train_loader, dev_loader = train.get_loaders(
+        train_set,
+        dev_set,
+        args.arch,
+        kwargs["batch_size"],
+        args.max_source_length,
+        args.max_target_length,
+    )
+    model = train.get_model(train_set, **kwargs)
+
+    # Train and log the best checkpoint.
+    best_checkpoint = train.train(
+        trainer, model, train_loader, dev_loader, args.train_from
+    )
+    util.log_info(f"Best checkpoint: {best_checkpoint}")
+
+    if args.predict:
+        # Deletes loaders to free those processes.
+        del train_loader
+        del dev_loader
+
+        # Reloads the dev set.
+        loader = predict.get_loader(
+            dataset=dev_set,
+            arch=args.arch,
+            batch_size=64,
+            max_source_length=defaults.MAX_SOURCE_LENGTH,
+            max_target_length=defaults.MAX_TARGET_LENGTH,
+        )
+        # Reloads the model.
+        model = predict.get_model(
+            arch=args.arch,
+            has_features=train_set.config.has_features,
+            checkpoint=best_checkpoint,
+        )
+        # Hack to get the results dir for this version from the checkpoint path
+        output = best_checkpoint.split("checkpoints")[0]
+        output = os.path.join(output, "predictions.tsv")
```

Comment:
Why not make this a path passed in by the `--predict` flag (like it is in `yoyodyne-predict`) rather than hard-code its path?

Reply:
I think the reason I did this is because we don't have the path to this exact version before runtime, and I had wanted them in the directory with metrics, etc. We can maintain a separate predictions dir, but then I am not sure how to map it to the other PTL path.

Reply:
I'm sorry but I don't understand this at all, can you explain?

Reply:
At runtime, PTL makes a path in `model_dir`/`experiment`/version_n, where version `n` is set at runtime. This is where PTL logs the `hparams.yml`, `metrics.csv` and all checkpoints. I made predict a boolean so that it defaults to writing to that same path.

Otherwise, I need to choose a path a-priori without knowing where the actual model contents will go.

Reply:
Ohhhh no. This is a file that writes our the predictions for dev on the best checkpoint after training. I think maybe I should just remove this, and I can keep it in my own local branch as a hack for convenience.